### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by usta**
* Update Dependencies and devDependencies
* Fix angular-file-upload restructured file system for assets in
defaults and production

### CODE REVIEW

**Comment by mleanos** [comment]
This duplicates what #1207 & #1213 are trying to accomplish. Is there something here that isn't covered in either of the two PR's?

File: package.json [hunk 1]
```diff
@@ -20,105 +20,105 @@
...
+    "body-parser": "~1.15.0",
+    "bower": "~1.7.7",
+    "cfenv": "~1.0.3",
+    "chalk": "^1.1.1",
+    "compression": "~1.6.1",
     "connect-flash": "~0.1.1",
-    "connect-mongo": "~0.8.1",
-    "consolidate": "~0.13.1",
-    "cookie-parser": "~1.3.2",
+    "connect-mongo": "~1.1.0",
```
Comment (mleanos) [inline package.json]: different than #1213

File: package.json [hunk 2]
```diff
@@ -20,105 +20,105 @@
...
+    "passport-facebook": "~2.1.0",
+    "passport-github": "~1.1.0",
+    "passport-google-oauth": "~1.0.0",
+    "passport-linkedin": "~1.0.0",
     "passport-local": "~1.0.0",
     "passport-paypal-openidconnect": "~0.1.1",
-    "passport-twitter": "~1.0.2",
-    "phantomjs": "~1.9.0",
+    "passport-twitter": "~1.0.4",
+    "phantomjs": "~2.1.3",
```
Comment (mleanos) [inline package.json]: different than #1213

File: package.json [hunk 3]
```diff
@@ -20,105 +20,105 @@
...
     "serve-favicon": "~2.3.0",
-    "socket.io": "~1.3.7",
+    "socket.io": "~1.4.5",
```
Comment (mleanos) [inline package.json]: different than #1213

**Comment by usta** [comment]
it combines both of them with more stable versions of dependencies

File: bower.json [hunk 4]
```diff
@@ -4,19 +4,19 @@
...
+    "angular": "~1.5.0",
+    "angular-animate": "~1.5.0",
+    "angular-bootstrap": "~1.1.2",
+    "angular-file-upload": "2.2.0",
+    "angular-messages": "~1.5.0",
+    "angular-mocks": "~1.5.0",
+    "angular-resource": "~1.5.0",
+    "angular-ui-router": "~0.2.18",
+    "angular-ui-utils": "0.1.1",
+    "bootstrap": "~3.3.6",
```
Comment (mleanos) [inline bower.json]: The previous three here are different than #1207

**Comment by mleanos** [comment]
The Angular dependency version's are defined better, in this PR. There have been issues with the tests passing with #1207 , so we'll see how this build does.

**Comment by mleanos** [comment]
It looks like the builds are failing. 

The Node `v0.12` is failing with the `mongo-connect` package; looks like it's using `const` which I don't think is compatible with `0.12`?

The Node 4 build is failing with the Karma test's, with a SocketIO error. I think we saw the same error in #1207.

**Comment by ilanbiala** [comment]
#1213 will handle server-side dependencies. We can do Bower stuff in this PR if you want.

**Comment by rhutchison** [comment]
Also duplicates #1198

**Comment by trendzetter** [comment]
Seems the blocking issue here is understanding what goes wrong in the tests. 
For my PR ( #1207 ) at least the functionality is working fully so it appears to me the tests need to be fixed.
The tests here are failing even worse than in the pr I proposed. Anyone having a grasp?

**Comment by ilanbiala** [comment]
@trendzetter I've had a couple school exams that I've needed to focus on, but hopefully today I will fix up mr PR.

**Comment by trendzetter** [comment]
@ilanbiala Good luck with the examinations and looking forward to seeing how this can be fixed.

**Comment by ilanbiala** [comment]
@usta let's just do client-side in this PR since #1213 does server-side.

**Comment by usta** [comment]
:) but this proofs that i need to improve my git skills :(

**Comment by usta** [comment]
karma still fails . It will be much better if we solve serverside dependencies problem before merging this. So this PR can wait until serverside part fixes (at least tests fixes )

**Comment by ilanbiala** [comment]
@usta you shouldn't have the node-inspector changes. Try rebasing onto master to get your changes to be applied properly.

**Comment by usta** [comment]
:) today i have learnt push --force :D  

by the way still i'm saying same thing """karma still fails . It will be much better if we solve serverside dependencies problem before merging this. So this PR can wait until serverside part fixes (at least tests fixes )"""

**Comment by ilanbiala** [comment]
@usta upgrading those dependencies requires fixing the code: https://travis-ci.org/meanjs/mean/jobs/110429226#L1880, these are all frontend errors so please take a look.

**Comment by usta** [comment]
@ilanbiala it looks like the problem caused because phantomjs's this version lack of bind function.
There is 2 solution : 1 - we might increase phantomjs to 2.1.3  2- we might add phantomjs-polyfill 
I will try both of solutions and try to fix this before merge

**Comment by lirantal** [comment]
@usta sounds better to just upgrade it.

**Comment by usta** [comment]
After upgrading phontomjs to "phantomjs-prebuilt": "~2.1.4" and  "karma-phantomjs-launcher": "~1.0.0", the only problem is about chat tests. chat test fails whenever used with use strict mode (socket.io.client.service.tests.js and chat.client.controller.tests.js )
I never good at tests, It looks like it will be much better to rewrite these 2 test

**Comment by usta** [comment]
@trendzetter  oopss it was a type, i will fix it also remove hardcoded resolutions

**Comment by usta** [comment]
I'm closing this PR to sake of #1207

**Pull request closed without merging**